src/map/utils/synthutils.cpp

```diff
@@ -659,77 +659,80 @@ namespace synthutils
 
     int32 doSynthFail(CCharEntity* PChar)
     {
-        uint8 currentCraft     = PChar->CraftContainer->getInvSlotID(0);
-        int16 synthDifficulty  = getSynthDifficulty(PChar, currentCraft);
-        int16 modSynthFailRate = PChar->getMod(Mod::SYNTH_FAIL_RATE);
+        if (PChar->CraftContainer->getCraftType() != CRAFT_NO_MATERIAL_LOSS) // If it's an unbreakable synth (Ex: Lu Shan), skip break calculations.
```

Comment:
Instead of wrapping everything in this new condition, you can see that the function only returns 0 (ie, the return is useless).

So instead you can avoid touching the whole rest of the function by returning early if its the new type:
```cpp
if (PChar->CraftContainer->getCraftType() == CRAFT_NO_MATERIAL_LOSS)
{
    return 0
}
```

Reply:
I thought about doing that at first, however, we still need the packet pushes at the end of the function, so if I were to return eatly, I would need to duplicate that logic.

Not oposing, just wondering if thats the best course of action.

Reply:
If its just a matter of sending one packet, that's fine to go inside the new chunk I'm suggesting, then the whole rest of the function doesn't have to be indented

Reply:
Its a bit more complex than a one liner.

I didnt feel comfortable with repeating all this code:
```c++
        // Push "Synthesis failed" messages.
        uint16 currentZone = PChar->loc.zone->GetID();

        if (currentZone &&
            currentZone != ZONE_MONORAIL_PRE_RELEASE &&
            currentZone != ZONE_49 &&
            currentZone < MAX_ZONEID)
        {
            PChar->loc.zone->PushPacket(PChar, CHAR_INRANGE, new CSynthResultMessagePacket(PChar, SYNTH_FAIL));
        }

        PChar->pushPacket(new CSynthMessagePacket(PChar, SYNTH_FAIL, 29695));
```

So instead I opted with separating the material loss logic into a new function called from the original function. I think that is a good compromise